federation/pkg/kubefed/init/init.go

```diff
@@ -173,6 +176,7 @@ func (o *initFederationOptions) Bind(flags *pflag.FlagSet, defaultServerImage, d
 	flags.Int32Var(&o.apiServerNodePortPort, apiserverPortFlag, 0, "Preferred port to use for api server nodeport service (0 for random port assignment). Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
 	flags.BoolVar(&o.apiServerEnableHTTPBasicAuth, "apiserver-enable-basic-auth", false, "Enables HTTP Basic authentication for the federation-apiserver. Defaults to false.")
 	flags.BoolVar(&o.apiServerEnableTokenAuth, "apiserver-enable-token-auth", false, "Enables token authentication for the federation-apiserver. Defaults to false.")
+	flags.StringVar(&o.tolerationsString, "tolerations", "", "If specified, the pod's tolerations. Example \"key1=value1:NoSchedule,key2=value2:NoSchedule\"")
```

Comment:
+1 for expose more parameters, also what do you mean by template, is it same as the issue https://github.com/kubernetes/kubernetes/issues/50817 ? @quinton-hoole